### DISCUSSION TITLE
How to load different data sources to generate columns？

### DISCUSSION THREAD

**Discussion by aprchen** [TanStack/table#4579] [Q&A]
I have some data like this, [{a:"test",b:"test1",c:1,...}...]

When I load data from a file, the data objects inside are not fixed, unlike the Person interface in the example, which only renders this data structure. In other words, I don't know what the data structure looks like until I read the data.
I use nextjs and ts, is there any example for reference?

**Comment by Tmcerlean** [comment]
I would also like to hear of any solutions to this problem
